src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -609,7 +611,10 @@ machine(MachineType:TCC, "TCC Cache")
     }
     if (coreRequestNetwork_in.isReady(clockEdge())) {
       peek(coreRequestNetwork_in, CPURequestMsg) {
-        if(in_msg.Type == CoherenceRequestType:RdBlk || in_msg.Type == CoherenceRequestType:Atomic){
+        if(in_msg.Type == CoherenceRequestType:RdBlk ||
+           in_msg.Type == CoherenceRequestType:Atomic ||
+           in_msg.Type == CoherenceRequestType:AtomicReturn ||
+           in_msg.Type == CoherenceRequestType:AtomicNoReturn){
           tbe.Destination.add(in_msg.Requestor);
         }
         tbe.isGLCSet := in_msg.isGLCSet;
```

Comment:
(2 lines down)  Do we need to check for AtomicReturn/AtomicNoReturn there too?

Reply:
Thinking about it more, I believe it is needed there as TCP can now generate two request types and that handles Atomics to same address.  I didn't find any others missing though, so that would be the last change

Reply:
I agree we need to check there.